service/history/ndc/workflow_resetter.go

```diff
@@ -349,23 +392,54 @@ func (r *workflowResetterImpl) persistToDB(
 		return nil
 	}
 
-	currentMutableState := currentWorkflow.GetMutableState()
-	currentRunID := currentMutableState.GetExecutionState().GetRunId()
-	currentCloseVersion, err := currentMutableState.GetCloseVersion()
+	// We have 3 different runs to update here - the base run, current run & the new run. But current is not yet closed.
+	if currentWorkflowMutation == nil {
+		return currentWorkflow.GetContext().ConflictResolveWorkflowExecution(
```

Comment:
workflowContext.ConflictResolveWorkflowExecution will trigger events reapply which we don't want I believe. Can just close base run as snapshot and call transaction.ConflictResolveWorkflowExecution.